controllers/provider-packet/pkg/webhook/controlplaneexposure/ensurer.go

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controlplaneexposure
+
+import (
+	"context"
+	"github.com/gardener/gardener-extensions/controllers/provider-packet/pkg/apis/config"
+	extensionscontroller "github.com/gardener/gardener-extensions/pkg/controller"
+	"github.com/gardener/gardener-extensions/pkg/webhook/controlplane"
+	"github.com/gardener/gardener-extensions/pkg/webhook/controlplane/genericmutator"
+
+	"github.com/gardener/gardener/pkg/operation/common"
+	"github.com/go-logr/logr"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// NewEnsurer creates a new controlplaneexposure ensurer.
+func NewEnsurer(etcdStorage *config.ETCDStorage, logger logr.Logger) genericmutator.Ensurer {
+	return &ensurer{
+		etcdStorage: etcdStorage,
+		logger:      logger.WithName("packet-controlplaneexposure-ensurer"),
+	}
+}
+
+type ensurer struct {
+	genericmutator.NoopEnsurer
+	etcdStorage *config.ETCDStorage
+	logger      logr.Logger
+}
+
+// EnsureETCDStatefulSet ensures that the etcd stateful sets conform to the provider requirements.
+func (e *ensurer) EnsureETCDStatefulSet(ctx context.Context, ss *appsv1.StatefulSet, cluster *extensionscontroller.Cluster) error {
+	e.ensureVolumeClaimTemplates(&ss.Spec, ss.Name)
+	return nil
+}
+
+func (e *ensurer) ensureVolumeClaimTemplates(spec *appsv1.StatefulSetSpec, name string) {
+	t := e.getVolumeClaimTemplate(name)
+	spec.VolumeClaimTemplates = controlplane.EnsurePVCWithName(spec.VolumeClaimTemplates, *t)
+}
+
+func (e *ensurer) getVolumeClaimTemplate(name string) *corev1.PersistentVolumeClaim {
+	var (
+		etcdStorage             config.ETCDStorage
+		volumeClaimTemplateName = name
+	)
+
+	if name == common.EtcdMainStatefulSetName {
+		etcdStorage = *e.etcdStorage
+		volumeClaimTemplateName = controlplane.EtcdMainVolumeClaimTemplateName
+	}
+
+	return controlplane.GetETCDVolumeClaimTemplate(volumeClaimTemplateName, etcdStorage.ClassName, etcdStorage.Capacity)
```

Comment:
You might want to check the Packet documentation if you can create 10Gi volumes of the `default` storage class (the default size that Gardener uses). If not, you need to specify a custom size here. See how it's done for Alicloud (the minimum size there turned out to be 20Gi)

Reply:
OK, it seems to be exactly 10Gi, https://github.com/packethost/csi-packet/search?q=MinVolumeSizeGi&unscoped_q=MinVolumeSizeGi